lesson24-Home-Work38/script/weather.js

```diff
@@ -0,0 +1,95 @@
+const api = {
+    key: "&units=metric&APPID=5d066958a60d315387d9492393935c19",
+    base: "http://api.openweathermap.org/data/2.5/weather?q="
+};
+
+const iconURL = "http://openweathermap.org/img/w/";
+
+const searchBtn = document.querySelector(".search-city-btn");
+const cityValue = document.querySelector(".search-box");
+const widget = document.querySelector(".container");
+const city = document.querySelector(".city");
+const date = document.querySelector(".date");
+const temperature = document.querySelector(".temp");
+const pressure = document.querySelector(".pressure");
+const weatherDescr = document.querySelector(".weather-description");
+const humidity = document.querySelector(".humidity");
+const speedWind = document.querySelector(".speed");
+const degWind = document.querySelector(".deg");
+const weatherPict = document.querySelector(".icon");
+const highLowDiv = document.querySelector(".hi-low");
+const error = document.querySelector(".error");
+const form = document.querySelector(".form");
+
+function getData(cityName) {
+    fetch(`${api.base}${cityName}${api.key}`)
```

Comment:
```
const API_ENDPOINT = "http://api.openweathermap.org/data/2.5/weather";

// ... url in fetch

const queryString = makeQueryStringFromObject(paramsObject)
const url = `${API_ENDPOINT}?${queryString}`;

fetch(url)
```

https://stackoverflow.com/questions/316781/how-to-build-query-string-with-javascript

https://code-boxx.com/build-query-string-javascript/

Reply:
```
const params = {
  parameter1: "value_1",
  parameter2: "value_2",
  parameter3: "value_3" 
};

const query = Object
  .keys(params)
  .map(k => encodeURIComponent(k) + '=' + encodeURIComponent(params[k]))
  .join('&');

console.log(query)
```

Reply:
const params = new URLSearchParams({
  parameter1: "value_1",
  parameter2: "value_2",
  parameter3: "value_3",
});

console.log(params.toString());